drivers/usbc/tcpc/ucpd_stm32.c

```diff
@@ -1374,6 +1377,37 @@ static int ucpd_init(const struct device *dev)
 	uint32_t cfg1;
 	int ret;
 
+#ifdef CONFIG_SOC_SERIES_STM32G0X
+	/* Configure CC pins as analog */
+	if ((uint32_t)config->ucpd_port == UCPD1_BASE) {
+		/* UCPD1_CC1 */
+		LL_GPIO_SetPinMode(GPIOA, LL_GPIO_PIN_8, LL_GPIO_MODE_ANALOG);
+		/* UCPD1_CC2 */
+		LL_GPIO_SetPinMode(GPIOB, LL_GPIO_PIN_15, LL_GPIO_MODE_ANALOG);
+	} else {
+		/* UCPD2_CC1 */
+		LL_GPIO_SetPinMode(GPIOD, LL_GPIO_PIN_0, LL_GPIO_MODE_ANALOG);
+		/* UCPD2_CC2 */
+		LL_GPIO_SetPinMode(GPIOD, LL_GPIO_PIN_2, LL_GPIO_MODE_ANALOG);
+	}
+#else
+	/* Disable Dead Dattery */
+	LL_PWR_DisableUCPDDeadBattery();
+#if CONFIG_SOC_SERIES_STM32G4X
+	/* Configure CC pins as analog */
+	/* UCPD1_CC1 */
+	LL_GPIO_SetPinMode(GPIOB, LL_GPIO_PIN_6, LL_GPIO_MODE_ANALOG);
+	/* UCPD1_CC2 */
+	LL_GPIO_SetPinMode(GPIOB, LL_GPIO_PIN_4, LL_GPIO_MODE_ANALOG);
+#else /* CONFIG_SOC_SERIES_STM32U5X || CONFIG_SOC_SERIES_STM32L5X */
+	/* Configure CC pins as analog */
+	/* UCPD1_CC1 */
+	LL_GPIO_SetPinMode(GPIOA, LL_GPIO_PIN_15, LL_GPIO_MODE_ANALOG);
+	/* UCPD1_CC2 */
+	LL_GPIO_SetPinMode(GPIOB, LL_GPIO_PIN_15, LL_GPIO_MODE_ANALOG);
+#endif /* CONFIG_SOC_SERIES_STM32G4X */
+#endif /* CONFIG_SOC_SERIES_STM32G0X */
```

Comment:
This doesn't scale as I expect PINs in use will vary across boards.
Please use dedicated pinctl API to do this.

Reply:
I've made the change and it's definitely much nicer.

Setting the CC pins as follows isn't so obvious: pinctrl-0 = <&analog_pb6 &analog_pb4>;
Could you add something like ucpd_cc1_pb6 and ucpd_cc2_pb4 to the modules/hal tx-pinctrl.dtsi?

Reply:
> Setting the CC pins as follows isn't so obvious: pinctrl-0 = <&analog_pb6 &analog_pb4>;
Could you add something like ucpd_cc1_pb6 and ucpd_cc2_pb4 to the modules/hal tx-pinctrl.dtsi?

Sure, will do.

Reply:
@sambhurst I've pushed https://github.com/zephyrproject-rtos/hal_stm32/pull/144.
To use it, you need to add a commit which will modify west.yml in the following way
```
    - name: hal_stm32
-      revision: 6253094a4147cbd30d2b466c975eb4013ac6be90
+      revision: pull/144/head
      path: modules/hal/stm32
      groups:
        - hal
```